src/qibolab/result.py

```diff
@@ -5,97 +5,186 @@
-        return phase
-        # return signal.detrend(np.unwrap(phase))
-
-    @cached_property
-    def ground_state_probability(self):
-        """Computes ground state probability"""
-        return 1 - np.mean(self.shots)
-
-    def raw_probability(self, state=1):
-        """Serialize probabilities in dict.
-        Args:
-            state (int): if 0 stores the probabilities of finding
-                        the ground state. If 1 stores the
-                        probabilities of finding the excited state.
-        """
-        if state == 1:
-            return {"probability": 1 - self.ground_state_probability}
-        elif state == 0:
-            return {"probability": self.ground_state_probability}
+        """Signal phase in radians."""
+        return np.angle(self.voltage.i + 1.0j * self.voltage.q)
+
+    # We are asumming results from the same experiment so same number of shots
+    def __add__(self, data):  # __add__(self, data:IntegratedResults) -> IntegratedResults
+        axis = 0
+        i = np.append(self.voltage.i, data.voltage.i, axis=axis)
+        q = np.append(self.voltage.q, data.voltage.q, axis=axis)
+        return IntegratedResults(i, q)
+
+    def serialize(self):
+        """Serialize as a dictionary."""
+        serialized_dict = {
+            "magnitude[V]": self.magnitude,
+            "i[V]": self.voltage.i,
+            "q[V]": self.voltage.q,
+            "phase[rad]": self.phase,
+        }
+        return serialized_dict
 
     @property
     def average(self):
         """Perform average over i and q"""
-        return AveragedResults.from_components(np.mean(self.i), np.mean(self.q))
+        average_i, average_q = np.array([]), np.array([])
+        std_i, std_q = np.array([]), np.array([])
+        for is_, qs_ in zip(self.voltage.i, self.voltage.q):
+            average_i, average_q = np.append(average_i, np.mean(is_)), np.append(average_q, np.mean(qs_))
+            std_i, std_q = np.append(std_i, np.std(is_)), np.append(std_q, np.std(qs_))
+        return AveragedIntegratedResults(average_i, average_q, std_i=std_i, std_q=std_q)
 
-    @property
-    def raw(self):
-        """Serialize output in dict."""
 
-        return {
-            "MSR[V]": self.measurement,
-            "i[V]": self.i,
-            "q[V]": self.q,
-            "phase[rad]": self.phase,
-        }
+# FIXME: Here I take the states from IntegratedResult that are typed to be ints but those are not what would you do ?
+class AveragedIntegratedResults(IntegratedResults):
+    """
+    Data structure to deal with the output of
+    :func:`qibolab.platforms.abstract.AbstractPlatform.execute_pulse_sequence`
+    :func:`qibolab.platforms.abstract.AbstractPlatform.sweep`
+
+    Associated with AcquisitionType.INTEGRATION and AveragingMode.CYCLIC
+    or the averages of ``IntegratedResults``
+    """
+
+    def __init__(self, i: np.ndarray, q: np.ndarray, shots=None, std_i=None, std_q=None):
```

Comment:
```suggestion
    def __init__(self, i: np.ndarray, q: np.ndarray, nshots=None, std_i=None, std_q=None):
```

I propose to use `nshots` instead of `shots` here and everywhere else, for better clarity

Reply:
sure